x/lscosmos/keeper/reset_data.go

```diff
@@ -21,234 +21,214 @@ const CHAIN_ID = "core-1"
+		// the current epoch has no unbondingEpochCvalue, so just add their stkATOMs
 		if delegatorUnbondingEntry.EpochNumber == currUnbondingEpoch {
 			totalStkAtomToBeInUnbondingAcc = totalStkAtomToBeInUnbondingAcc.Add(delegatorUnbondingEntry.Amount.Amount)
 			continue
 		}
-		//
-		unbondingEpochCValue, found := findUnbondingCValue(allUnbondingEpochCValues, delegatorUnbondingEntry.EpochNumber)
+
+		// retrieve the unbonding c value for the delegator unbonding entry
+		unbondingEpochCValue, found := k.findUnbondingCValue(ctx, delegatorUnbondingEntry.EpochNumber)
 		if !found {
 			// should never occur as per the code logic, but if not found, just continue.
-			k.Logger(ctx).Error("Unbonding epoch not found for ", "epochnumber", delegatorUnbondingEntry.EpochNumber, "delegatorunbondingentry", delegatorUnbondingEntry)
+			k.Logger(ctx).Error(
+				"Unbonding epoch not found",
+				"epoch_number",
+				delegatorUnbondingEntry.EpochNumber,
+				"delegator_unbonding_entry",
+				delegatorUnbondingEntry,
+			)
 			continue
 		}
-		// if is matured, their claims are atoms, not stkatoms.
+
+		// if the unbonding c value is matured, the claim will be in ATOM, so nothing to do
 		if unbondingEpochCValue.IsMatured {
 			continue
 		}
-		hostAccUnbonding, found := findHostAccUnbonding(delegationState.HostAccountUndelegations, delegatorUnbondingEntry.EpochNumber)
-		//filter the unbonding requests we might have stored that are not in the unstaking period (should filter 164,216,228 epochs)
+
+		// get the host account unbonding for the epoch
+		hostAccUnbonding, found := k.findHostAccUnbonding(ctx, delegatorUnbondingEntry.EpochNumber)
+
+		// filter the unbonding requests that are in the maturing state but have no maturing time set
+		// this should filter out [164, 216, 228] epochs
 		if !unbondingEpochCValue.IsFailed && found && hostAccUnbonding.CompletionTime.Equal(time.Time{}) {
 			totalStkAtomToBeInUnbondingAcc = totalStkAtomToBeInUnbondingAcc.Add(delegatorUnbondingEntry.Amount.Amount)
 			continue
 		}
-		// if unstaking is failed, their stkatoms should be available for claim,
-		//NOTE the reset of Db has to occur before this function orelse 232 will come here
+
+		// if the unbonding has failed, the stkATOMs need to be claimed, so we can add the amount
+		// NOTE: the re-creation of the deleted epochs needs to happen before this
 		if unbondingEpochCValue.IsFailed {
 			totalStkAtomToBeInUnbondingAcc = totalStkAtomToBeInUnbondingAcc.Add(delegatorUnbondingEntry.Amount.Amount)
 			continue
 		}
 	}
+
 	return totalStkAtomToBeInUnbondingAcc
 }
 
-func findUnbondingCValue(allunbondingCValues []types.UnbondingEpochCValue, epoch int64) (types.UnbondingEpochCValue, bool) {
-	for _, unbondingEpochCvalue := range allunbondingCValues {
+func (k Keeper) findUnbondingCValue(ctx sdk.Context, epoch int64) (types.UnbondingEpochCValue, bool) {
+	for _, unbondingEpochCvalue := range k.IterateAllUnbondingEpochCValues(ctx) {
```

Comment:
won't this consume more gas? I think it's better to just query once and then pass.